docs/modules/ROOT/pages/faq.adoc

```diff
@@ -218,26 +217,90 @@ Renaming a variable is disallowed by default because there is a chance that a re
 
 It is possible to disable this check by passing the option `unsafeAllowRenames: true`. A more granular approach is to use a docstring comment `/// @custom:oz-renamed-from <previous name>` right above the variable that is being renamed, for example:
 
-```
+[source,solidity]
+----
 contract V1 {
     uint x;
 }
 contract V2 {
     /// @custom:oz-renamed-from x
     uint y;
 }
-```
+----
 
 Changing the type of a variable is not allowed either, even in cases where the types have the same size and alignment, for the similar reason explained above. As long as we can guarantee that the rest of the layout is not affected by this type change, it is also possible to override this check by placing a docstring comment `/// @custom:oz-retyped-from <previous type>`.
 
-```
+[source,solidity]
+----
 contract V1 {
     bool x;
 }
 contract V2 {
     /// @custom:oz-retyped-from bool
     uint8 x;
 }
-```
+----
 
 Docstring comments don't yet work for struct members, due to a current Solidity limitation.
+
+[[internal-function-storage]]
+== How can I use internal functions in storage variables?
+
+Internal functions in storage variables are code pointers which will no longer be valid after an upgrade, because the code will move around and the pointer would change. To avoid this issue, you can either declare those functions as external, or reassign internal functions during upgrades.
```

Comment:
Honestly this would probably be better practice:

```suggestion
Internal functions in storage variables are code pointers which will no longer be valid after an upgrade, because the code will move around and the pointer would change. To avoid this issue, you can either declare those functions as external, or reassign internal functions during upgrades. Alternatively, avoid code pointers in storage altogether, and define an `enum` that you will use to select with a dispatcher function to select from the list of available functions.
```

Note sure if the description is clear enough like this.

Reply:
Is there any reason for the specific recommendation to use an `enum`, or would any implementation of a dispatcher function be suitable for this scenario?

Reply:
An enum is just the classic way you would implement this and I think it's clean, but any type would work!